### PR TITLE
CAMEL-23703: Package-native validation CI for the Camel CLI launcher

### PR DESCRIPTION
**Pull request by ammachado**
# Description

Adds CI that validates the Camel CLI launcher distribution the way real package managers consume it, plus the installer/manifest hardening that validation depends on.

**CI workflow** (`.github/workflows/package-native-validation.yml`)
- Builds the launcher distribution and validates it on macOS + Linux (POSIX validators) and Windows (installer unit tests), triggered on changes under `dsl/camel-jbang/camel-launcher/**` and `tooling/camel-exe/**`.
- Stages a synthetic (non-`SNAPSHOT`) release version offline so `camel-package.sh prepare` runs without publishing anything, then runs the validators host-gated.

**POSIX validators** (`src/jreleaser/bin/camel-validate.sh` + `lib/assert-camel-cli.sh`)
- A dispatcher that validates the local archive directly, plus host-gated Homebrew (audit `--strict` + install/version/init/uninstall via a throwaway local tap, with an offline `file://` formula rewrite in test mode) and SDKMAN (descriptor + archive-structure checks) legs. Any genuine failure exits nonzero; absent package managers `SKIP`.

**Installer + manifest hardening**
- `install.sh` / `install.ps1` tolerate `#`/`##` comment lines in the release manifest (so the manifest can carry an ASF license header) while still rejecting blank lines and enforcing the four required keys.
- `install.sh` surfaces `wget`/`tar` errors instead of silencing them (`-nv`, captured stderr).
- `WebsiteManifestGenerator` prepends an ASF license header to generated manifests; its strict re-parser and both installers skip the comment lines.
- `install.ps1` now works correctly under PowerShell 7 (`pwsh`), not just Windows PowerShell: the test-seam certificate trust switches from the .NET Framework-only `ServicePointManager.ServerCertificateValidationCallback` (silently ignored by PS7's `Invoke-WebRequest`) to `-SkipCertificateCheck` on Core edition, and the archive entry Unix-mode check no longer throws on any entry with mode bits set (a checked `[uint32]` cast on a value that legitimately overflows Int32 range).

**Docs**: "Installing the Camel CLI Launcher" guide (merged with the newer `camel self-update`/`camel doctor` content already on `main`), plus nav / launcher-page / upgrade-guide updates.

**Tests**: `PackageNativeValidationTest` (host-gated orchestration of the validators and assertion library), installer license-header tolerance tests, manifest generator tests, and an `expected-init-route.txt` fixture.

### Rebuilt onto current `main` (this push)

`main` picked up an overlapping change (#25072, `camel self-update`/`camel doctor`) that touched most of the same files this branch does, so a literal rebase conflicted on nearly every commit. The branch was reconstructed from scratch on top of current `main`'s 3 commits instead of replayed commit-by-commit; net content is the same, history is not. While reconciling, two real issues surfaced and are fixed here:

- **Stale Homebrew formula naming.** `main`'s #25072 renamed the generated formula from `camel`/`camel@<line>` to `apache-camel`/`apache-camel@<line>`. This branch's `camel-validate.sh` and `PackageNativeValidationTest`'s `stageHomebrewFormula()` fixture still referenced the old name in a couple of spots reintroduced during reconciliation; both now match `main`'s renamed formula.
- **`windows-validator`'s 600s fork timeout (root cause identified and fixed).** The `windows-validator` CI job on this PR's prior run (before this push) was killed by the module's surefire fork timeout with zero tests reported complete. That run included temporary `[diag]` timing instrumentation added specifically to investigate it; the captured timings showed no hang — each `WebsiteInstallTest$WindowsPowerShell` test costs a near-constant ~24-25s, almost entirely spent in a single `install.ps1` invocation via a fresh `powershell.exe` (PowerShell startup/Defender-scan overhead on `windows-latest` runners). With ~25 tests in the class, cumulative time (~625s) runs past the 600s budget even though no individual subprocess call is anywhere near its own 30s timeout. Fixed by raising `camel.surefire.forkTimeout` to `1200` in `camel-launcher/pom.xml`, the same fix already applied to `camel-kafka`, `camel-jms`, and `camel-ftp` for the identical symptom. The diagnostics instrumentation has been removed now that it served its purpose.

### Verification
- `dsl/camel-jbang/camel-launcher` unit suite: 221 run, 0 failures (35 OS-gated skips), re-verified after the reconciliation and again after the fork-timeout fix.
- `camel-package.sh prepare` reproduced offline end to end: it exits 0 and generates the Homebrew formula the validator consumes.
- Full `mvnd clean install -Dquickly` from the repo root passes.
- The `install.ps1` PS7 fixes and the fork-timeout root cause were both confirmed against a real `windows-validator` CI run on this PR (see above), not just local reasoning.

# Target

- [x] I checked that the commit is targeting the correct branch (Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it). — [CAMEL-23703](https://issues.apache.org/jira/browse/CAMEL-23703)

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

  <sub>`mvnd clean install -Dquickly` passed from the repo root and the launcher module's own test suite is green; a non-`-Dquickly` full-root build has not been re-run for this push.</sub>

# AI-assisted contributions

- [x] If this PR includes AI-generated code, commits have proper co-authorship attribution (e.g., `Co-authored-by` trailers) and the PR description identifies the AI tool used.

  <sub>All three commits on this branch (the reconciled package-native-validation-CI addition, the Homebrew formula-naming fix, and the fork-timeout increase) were AI-assisted with Claude Code (Sonnet 5) and carry `Co-authored-by` trailers. _Claude Code on behalf of ammachado._</sub>

### CODE REVIEW

File: .github/workflows/package-native-validation.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            echo "/home/linuxbrew/.linuxbrew/sbin"
+          } >> "$GITHUB_PATH"
+          {
+            echo "HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew"
+            echo "HOMEBREW_CELLAR=/home/linuxbrew/.linuxbrew/Cellar"
+            echo "HOMEBREW_REPOSITORY=/home/linuxbrew/.linuxbrew/Homebrew"
+          } >> "$GITHUB_ENV"
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@be666c2fcd27ec809703dec50e508c2fdc7f6654 # v5.2.0
```
Comment (apupier) [inline .github/workflows/package-native-validation.yml]: 5.6.0 is available https://github.com/actions/setup-java/releases/tag/v5.6.0

File: .github/workflows/package-native-validation.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+jobs:
+  # ── macOS / Linux (POSIX validators) ──────────────────────────────
+  posix-validator:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    steps:
+      - uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
```
Comment (apupier) [inline .github/workflows/package-native-validation.yml]: 7.0.1 is available https://github.com/actions/checkout/releases/tag/v7.0.1

File: .github/workflows/package-native-validation.yml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        shell: bash
+        run: |
+          cd dsl/camel-jbang/camel-launcher
+          mvn -B -ntp test -pl . -Dtest=PackageNativeValidationTest
+
+  # ── Windows (install.ps1 unit tests) ──────────────────────────────
+  windows-validator:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
```
Comment (apupier) [inline .github/workflows/package-native-validation.yml]: 7.0.1 is available https://github.com/actions/checkout/releases/tag/v7.0.1

File: .github/workflows/package-native-validation.yml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        with:
+          persist-credentials: false
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@be666c2fcd27ec809703dec50e508c2fdc7f6654 # v5.2.0
```
Comment (apupier) [inline .github/workflows/package-native-validation.yml]: 5.6.0 is available

**Review by apupier — DISMISSED** [review]
use latest version of github actions

not reviewed the other parts

**Review by davsclaus — COMMENTED** [review]
## Review Summary

_Claude Code on behalf of davsclaus_

No blocking issues found. The PR is well-crafted with solid test coverage and documentation.

### Confirmed Issues (minor)

1. **`actions/setup-java` version inconsistency across workflows** — This PR uses `setup-java@v5.6.0`, while the rest of the repo (`pr-build-main.yml`, `main-build.yml`, etc.) uses `v5.2.0`. Not a bug — v5.6.0 is newer and correct — but creates version drift. Ideally either this PR or a follow-up should align the other workflows.

2. **Build checklist item partially satisfied** — The PR checklist notes that `mvnd clean install -Dquickly` was run instead of the full `mvn clean install -DskipTests`. Since this PR doesn't modify component metadata/APIs, auto-generated file changes are unlikely, but worth confirming.

### Observations

- **Stale CHANGES_REQUESTED from apupier (July 22)**: The current push already uses the action versions requested (`checkout@v7.0.1`, `setup-java@v5.6.0`). The review status appears resolved.
- **Comprehensive test coverage**: 17 tests covering happy paths, error propagation, edge cases (SNAPSHOT version tolerance, stderr surfacing), input validation, and host-gating behavior.
- **Follows project conventions**: AssertJ assertions, package-private test class/methods, no `Thread.sleep()`, records for internal types, proper JIRA reference in commits.
- **Documentation updated**: Upgrade guide entry, installer docs (version pinning, requirements, uninstall), and nav.adoc all updated.
- **PowerShell 7 fix is correct**: The PS7 compatibility issue (`ServicePointManager.ServerCertificateValidationCallback` silently ignored by `HttpClient`-backed `Invoke-WebRequest`) is a real bug with a correct fix.
- **Surefire timeout increase justified**: Backed by actual CI timing data, consistent with identical increases already applied to `camel-kafka`, `camel-jms`, `camel-ftp`.

### Not in scope

This review covers project rules, conventions, and structural correctness. It does not replace specialized AI review tools (CodeRabbit, Sourcery) or static analyzers (SonarCloud).

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

**Comment by davsclaus** [comment]
> actions/setup-java version inconsistency across workflows — This PR uses setup-java@v5.6.0, while the rest of the repo (pr-build-main.yml, main-build.yml, etc.) uses v5.2.0. Not a bug — v5.6.0 is newer and correct — but creates version drift. Ideally either this PR or a follow-up should align the other workflows.

We should upgrade these others

**Review by davsclaus — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
LGTM

**Review by davsclaus — APPROVED** [review]

**Review by gnodet — APPROVED** [review]
Thorough CI workflow for offline package-native validation of the Camel CLI launcher. Well-structured and comprehensive.

**Highlights:**
- Proper workflow conventions: read-only permissions, `persist-credentials: false`, concurrency cancellation, path-based triggers, no Docker Hub images
- PowerShell 7 compatibility fix is technically correct — `ServicePointManager.ServerCertificateValidationCallback` is silently ignored by PS7's `HttpClient`-based `Invoke-WebRequest`; the compiled-type callback via `Add-Type` for PS 5.1 correctly avoids the Runspace threading issue
- `SharedTls` double-checked locking (`volatile` + `synchronized`) is correctly implemented
- Test class follows all project conventions: package-private, AssertJ assertions, no `Thread.sleep()`, proper `@TempDir` usage, Records for internal types, `@DisabledOnOs` for platform gating
- Surefire fork timeout increase (600→1200s) is well-justified by measured per-test timings on Windows runners
- Homebrew formula fixes address real `brew audit --strict` requirements
- Documentation updates (upgrade guide, install page, nav) are present and use `xref:` for internal links

**Minor observations (non-blocking):**
- `actions/setup-java` v5.6.0 here vs v5.2.0 in other workflows creates drift (already noted by davsclaus)
- `MODULE_DIR` in `PackageNativeValidationTest` actually resolves to `src/` rather than the module root — `SRC_DIR` would be slightly more accurate

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

_Claude Code on behalf of Guillaume Nodet_
